### PR TITLE
Refactor API

### PR DESCRIPTION
**Pull request by romgrk**
Refactor API and make its use less verbose.

Note: this transformation can also be done in other places, I leave it to you to do that.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -40,12 +40,7 @@ const App = () => {
 
   // load Beacon name and a few statistics
   useEffect(() => {
-    const getBeaconInfo = async () => {
-      const results = await apiGet("/info");
-      console.log({ returnedGet: results.response });
-      setBeaconInfo(parseBeaconInfo(results.response));
-    };
-    getBeaconInfo();
```
Comment (romgrk) [inline src/App.js:48]: This was a bit too verbose for what it does.

File: src/api.js [hunk 2]
```diff
@@ -1,40 +1,31 @@
 
-// note: api base url set in package.json proxy
-
-// start with separate GET and POST calls with minimal error handling, 
-// I can make it pretty later
```
Comment (romgrk) [inline src/api.js:5]: > // I can make it pretty later

Bad idea.

File: src/api.js [hunk 3]
```diff
@@ -1,40 +1,31 @@
...
-      headers: {
-        'Content-Type': 'application/json'
-      },
-    })
-      .then((res) => res.json())
-      .then((json) => {  
-        console.log({ apiGetResults: json });
-        return json;
-      })
-      .catch((err) => console.log(err)))
```
Comment (romgrk) [inline src/api.js:19]: Don't catch errors if you don't handle them properly. Let them propagate.

**Comment by gsfk** [comment]
nice, I like the verbosity reduction

**Pull request merged**